src/client/Microsoft.Identity.Client/AuthenticationInfoParameters.cs

```diff
@@ -0,0 +1,90 @@
+    {
+        private const string AuthenticationInfoKey = "Authentication-Info";
+        /// <summary>
+        /// The next nonce to be used in the preceding authentication request.
+        /// </summary>
+        public string NextNonce { get; private set; }
+
+        /// <summary>
+        /// Return the <see cref="RawParameters"/> of key <paramref name="key"/>.
+        /// </summary>
+        /// <param name="key">Name of the raw parameter to retrieve.</param>
+        /// <returns>The raw parameter if it exists,
+        /// or throws a <see cref="System.Collections.Generic.KeyNotFoundException"/> otherwise.
+        /// </returns>
+        public string this[string key]
+        {
+            get
+            {
+                return RawParameters[key];
+            }
+        }
+
+        /// <summary>
+        /// Dictionary of raw parameters in the Authentication-Info header (extracted from the Authentication-Info header
+        /// string value, without any processing). This allows support for APIs which are not mappable easily to the standard
+        /// or framework specific (Microsoft.Identity.Model, Microsoft.Identity.Web).
+        /// </summary>
+        internal IDictionary<string, string> RawParameters { get; private set; }
+
+        /// <summary>
+        /// Create Authentication-Info parameters from the HttpResponseHeaders for each auth scheme.
+        /// </summary>
+        /// <param name="httpResponseHeaders">HttpResponseHeaders.</param>
+        /// <returns>Authentication-Info provided by the endpoint</returns>
+        public static AuthenticationInfoParameters CreateFromHeaders(HttpResponseHeaders httpResponseHeaders)
+        {
+            AuthenticationInfoParameters parameters = new AuthenticationInfoParameters();
+
+            try
+            {
+                if (httpResponseHeaders.Contains(AuthenticationInfoKey))
+                {
+                    var authInfoValue = httpResponseHeaders.Where(header => header.Key == AuthenticationInfoKey).Single().Value.FirstOrDefault();
+
+
+                    var AuthValuesSplit = authInfoValue.Split(new char[] { ' ' }, 2);
+
+                    var paramValues = CoreHelpers.SplitWithQuotes(AuthValuesSplit[1], ',')
+                            .Select(v => AuthenticationHeaderParser.CreateKeyValuePair(v.Trim(), AuthenticationInfoKey))
+                            .ToDictionary(pair => pair.Key, pair => pair.Value, StringComparer.OrdinalIgnoreCase);
+
+                    parameters.RawParameters = paramValues;
+
+                    if (paramValues.TryGetValue("nextnonce", out string value))
+                    {
+                        parameters.NextNonce = value;
+                    }
+
+                    return parameters;
+
+                    //Could not get Auth info parameters
+                    throw new MsalClientException(MsalError.UnableToParseAuthenticationHeader, MsalErrorMessage.UnableToParseAuthenticationHeader);
```

Comment:
If header is not present, you should behave like WWW-Authenticate parser, which doesn't throw. Note: this is a blocking comment